File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int maxcur=0;
        int maxi = 0;
        for(int i=1;i<n;i++){
            maxcur = max(0, maxcur+=prices[i]-prices[i-1]);
            maxi = max(maxi,maxcur);
        }
        return maxi;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        small = prices[0]
        profit = 0
        maxi = 0
        for i in range(len(prices)):
            if prices[i]<small:
                small = prices[i]
            profit = prices[i] - small
            maxi = max(maxi,profit)
        return maxi        
        